src/app.ts

```diff
@@ -48,28 +51,68 @@ const redisOptions = <RedisOptions>{
     host: config.RedisHost,
     port: config.RedisPort,
 };
-const redisQueueOptions = <ClientOpts>{
-    host: config.RedisHost,
-    port: config.RedisPort,
-};
 
 if (config.RedisPassword) {
     redisOptions.password = config.RedisPassword;
-    redisQueueOptions.password = config.RedisPassword;
 }
 
 if (config.RedisTLS) {
     redisOptions.tls = {};
-    redisQueueOptions.tls = {};
 }
 
 if (config.RedisDb) {
     redisOptions.db = config.RedisDb;
-    redisQueueOptions.db = config.RedisDb;
 }
 
 const redisClient = new Redis(redisOptions);
-const bareRedisClient = new RedisClient(redisQueueOptions);
+
+let metricsStore: MetricsStore;
+
+switch (config.MetricsStoreProvider) {
+    case 'prometheus':
+        metricsStore = new PrometheusClient({
+            logger,
+            endpoint: config.PrometheusURL,
+        });
+        break;
+    default:
+    case 'redis':
```

Comment:
This statement is unnecessary, if we want the default to be redis. Just keep the default and perhaps add a comment which says `// redis`